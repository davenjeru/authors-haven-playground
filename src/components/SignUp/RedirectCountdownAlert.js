import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import React from 'react';

/** *This is the alert that will show the user that they will be redirected after a few seconds
 * @param secondsToRedirect: {number} The seconds remaining before redirecting */
const RedirectCountdownAlert = ({ secondsToRedirect }) => (
  <div
    className="sign-up-success-message alert alert-warning text-center"
    style={{ marginTop: '5em' }}
  >
    We will redirect you to the
    {' '}
    <Link to="/login">login</Link>
    {' '}
    page in
    {' '}
    <strong>{secondsToRedirect}</strong>
    {' '}
    seconds
  </div>
);

RedirectCountdownAlert.propTypes = {
  secondsToRedirect: PropTypes.number.isRequired,
};

export default RedirectCountdownAlert;
