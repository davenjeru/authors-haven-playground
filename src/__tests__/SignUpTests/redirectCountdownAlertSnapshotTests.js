import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import RedirectCountdownAlert from '../../components/SignUp/RedirectCountdownAlert';

describe('The SignUpFormCard', () => {
  it('should not regress', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <RedirectCountdownAlert secondsToRedirect={10} />
      </MemoryRouter>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
