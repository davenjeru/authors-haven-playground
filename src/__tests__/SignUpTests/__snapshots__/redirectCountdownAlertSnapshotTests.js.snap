// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignUpFormCard should not regress 1`] = `
<div
  className="sign-up-success-message alert alert-warning text-center"
  style={
    Object {
      "marginTop": "5em",
    }
  }
>
  We will redirect you to the
   
  <a
    href="/login"
    onClick={[Function]}
  >
    login
  </a>
   
  page in
   
  <strong>
    10
  </strong>
   
  seconds
</div>
`;
